spec/requests/api/topics/index_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe 'Topics', type: :request do
+  shared_examples 'success response' do
+    it 'returns a successful response' do
+      get_topics
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  subject(:get_topics) do
+    get api_topics_path, as: :json
+  end
```

Comment:
```suggestion
  subject(:get_topics) { get api_topics_path, as: :json }
```
you can do this in one line if you want to